build/build.go

```diff
@@ -187,6 +187,23 @@ func ImportDir(dir string, mode build.ImportMode, installSuffix string, buildTag
 	return &PackageData{Package: pkg, JSFiles: jsFiles}, nil
 }
 
+// Test if we find the '//gopherjs:keep_overridden' comment
+func findKeepOverridenComment(doc *ast.CommentGroup) bool {
+	if doc == nil {
+		return false
+	}
+	for _, comment := range doc.List {
+		text := comment.Text
```

Comment:
Might I suggest the following is slightly clearer in its intention?

```go
text := strings.TrimRightFunc(comment.Text, unicode.IsSpace)
if text == "//gopherjs:keep_overridden" {
  return true
}
```

i.e. you're looking for a line that contains only the special comment, save any trailing space.

Reply:
I actually took this part from the go pragma directive parser:
https://github.com/golang/go/blob/master/src/cmd/compile/internal/gc/noder.go#L1435
It seems like you can add text after a go pragma directive, and it will be ignored, and I think it's a good idea to do the same here.

Reply:
Got it. I think you'll find that is effectively the parse phase of the compiler; so it has to deal with all types of pragma. [Many of which take arguments](https://gist.github.com/myitcv/e4202a9513bed098262444ed0f4f3625).

In this case we've already parsed into the AST and we know we have a comment group.

And specifically, I'm guessing you want `//gopherjs:keep_overridden` to be used without arguments, because there are none. Hence it doesn't make sense for anything to follow that special comment.

Reply:
Considering the [pragma function from the standard library]( https://github.com/golang/go/blob/20c98dc396d1cb8c3a9776f1d330feed2f2a132a/src/cmd/compile/internal/gc/noder.go#L1397), my understanding is that pragmas expecing arguments (`go:linkname`, `go:cgo_*`) are treated in the non-default cases of the switch. The default case, from which this code is taken, only treats pragmas with no arguments (`go:nointerface`, `go:noescape`, etc., see the implementation of [pragmaValues](https://github.com/golang/go/blob/20c98dc396d1cb8c3a9776f1d330feed2f2a132a/src/cmd/compile/internal/gc/lex.go#L52))
So it looks like the standard library allows putting any text (so basically a comment) after a pragma without argument. I think it's appropriate to have the same behaviour here (even if I agree that it's not the typical use case).

Reply:
> So it looks like the standard library allows putting any text (so basically a comment) after a pragma without argument. I think it's appropriate to have the same behaviour here (even if I agree that it's not the typical use case).

You're absolutely right, thanks for correcting me.

I'm easy either way; in this PR I think you can be very specific and prescriptive because you're defining the pragma so it's in your gift to do what you want 😄 - to my mind doing so keeps the documentation (on how to use your pragma) simple as well as the implementation. 

But, as I said earlier, it's a minor point in any case.